tools/mount.cpp

```diff
@@ -226,19 +226,39 @@ int main(int argc, char** argv)
     {
         const char* source = argv[2];
         struct stat sb;
-        if (stat(source, &sb) != 0 || !S_ISDIR(sb.st_mode))
+        if (stat(source, &sb))
         {
-            fprintf(stderr, "%s: cannot mount from invalid source directory [%s].\n", program,
-                    source);
+            fprintf(stderr, "%s: cannot mount from invalid source [%s]. stat failed with %s\n",
+                    program, source, strerror(errno));
+            return EX_USAGE;
+        }
+
+        const bool isDir = S_ISDIR(sb.st_mode);
+        const bool isFile = S_ISCHR(sb.st_mode); // We don't support regular files.
```

Comment:
So - this is security related; lets special-case this for bind mounting files only from /dev/[u]random to wherever - we want to have the narrowest window possible in this CAP_SYSADMIN code.

Reply:
Yes, noted.